docs/managing-dragonfly/replication.md

```diff
@@ -39,3 +39,88 @@ replicaof no one
 
 This will stop the instance from replicating the primary server but will not discard the dataset it has already replicated. This syntax is useful in cases where the original primary fails. After running `replicaof no one` on a replica of the failed primary, the former replica can be used as the new primary and have its own replicas as a failsafe.
 
+
+## Secure replication with TLS -- Prerequisites
+
+Dragonfly supports replication over TLS. To make it work you need:
+
+1. A valid certificate issued by a CA. We recommend [let's encrypt](https://letsencrypt.org/).
+2. An x509 server certificate and private key.
+3. An x509 client certificate and private key.
+
+Also note that we only support `.pem` files, any other format won't work with dragonfly.
+
+For the second and third step you can use `OpenSSL` lib to generate the required files. For example, the process of generating a server certificate is:
+
+Generate Dragonfly's private key and certificate signing request (CSR):
+```bash
+openssl req -newkey rsa:4096 -nodes -keyout server-key.pem -out server-req.pem
+```
+This will prompt you to add additional details for the subject which contains information
+related to the certificate. The attributes of the subject are key-value pairs and are the following:
+
+- CN: CommonName
+- OU: OrganizationalUnit
+- O: Organization
+- L: Locality
+- S: StateOrProvinceName
+- C: CountryName
+
+The `-nodes` part means `no DES`, that is, do not encrypt the private key. For production use cases,
+you should always consider encrypting private keys.
+
+Finally, you should use CA's private key to sign dragonfly's CSR and get back the signed certificate
+```bash
+openssl x509 -req -in server-req.pem -days 365 -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out server-cert.pem
+```
+
+The `ca-cert.pem` and `ca-key.pem` are retrieved on the first step described above.
+
+The same process applies for generating client keys.
+
+## Secure replication with TLS
+
+To enable TLS for dragonfly, you must supply the following arguments:
+
+```bash
+./dragonfly --tls --tls_key_file=server-key.pem --tls_cert_file=server-cert.pem --tls_ca_cert_file=ca-cert.pem --tls_replication=true
+```
+
+Start the replica on a different port (in this example the port is 6380, and the host is local host):
+```bash
+./dragonfly --tls --tls_key_file=server-key.pem --tls_cert_file=server-cert.pem --tls_ca_cert_file=ca-cert.pem --tls_replication=true --port 6380
```

Comment:
I think its better to give an example where master and replica dont run on the same vm
use port 6379 for both master and replica